python310Packages.prometheus-client: 0.15.0 -> 0.16.0

vamega opened the pull request:
###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

Commit:
pythnon310Packages.prometheus-client: 0.15.0 -> 0.16.0

vamega commented:
Result of `nixpkgs-review pr 215880` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>53 packages failed to build:</summary>
  <ul>
    <li>dbx</li>
    <li>ihaskell</li>
    <li>jupyter</li>
    <li>manim</li>
    <li>python310Packages.Nikola</li>
    <li>python310Packages.ansible-kernel</li>
    <li>python310Packages.arviz</li>
    <li>python310Packages.augmax</li>
    <li>python310Packages.bambi</li>
    <li>python310Packages.batchspawner</li>
    <li>python310Packages.dalle-mini</li>
    <li>python310Packages.dockerspawner</li>
    <li>python310Packages.einops</li>
    <li>python310Packages.elegy</li>
    <li>python310Packages.jupyter</li>
    <li>python310Packages.jupyter-lsp</li>
    <li>python310Packages.jupyter-server</li>
    <li>python310Packages.jupyter-server-mathjax</li>
    <li>python310Packages.jupyterhub</li>
    <li>python310Packages.jupyterhub-ldapauthenticator</li>
    <li>python310Packages.jupyterhub-systemdspawner</li>
    <li>python310Packages.jupyterhub-tmpauthenticator</li>
    <li>python310Packages.jupyterlab</li>
    <li>python310Packages.jupyterlab-git</li>
    <li>python310Packages.jupyterlab-lsp</li>
    <li>python310Packages.jupyterlab_launcher</li>
    <li>python310Packages.jupyterlab_server</li>
    <li>python310Packages.jupytext</li>
    <li>python310Packages.mkdocs-jupyter</li>
    <li>python310Packages.nbclassic</li>
    <li>python310Packages.nbdime</li>
    <li>python310Packages.nix-kernel</li>
    <li>python310Packages.notebook</li>
    <li>python310Packages.notebook-shim</li>
    <li>python310Packages.notedown</li>
    <li>python310Packages.oauthenticator</li>
    <li>python310Packages.phik</li>
    <li>python310Packages.pymc</li>
    <li>python310Packages.pymoo</li>
    <li>python310Packages.treex</li>
    <li>python310Packages.vector</li>
    <li>python310Packages.vega</li>
    <li>python310Packages.xarray-einstats</li>
    <li>python311Packages.ansible-kernel</li>
    <li>python311Packages.cairo-lang</li>
    <li>python311Packages.gradient</li>
    <li>python311Packages.jupyter-server-mathjax</li>
    <li>python311Packages.jupyterlab-git</li>
    <li>python311Packages.nbdime</li>
    <li>quarto</li>
    <li>sage</li>
    <li>sageWithDoc</li>
    <li>wolfram-notebook</li>
  </ul>
</details>
<details>
  <summary>81 packages built:</summary>
  <ul>
    <li>datadog-agent</li>
    <li>datadog-process-agent</li>
    <li>dmarc-metrics-exporter</li>
    <li>home-assistant-component-tests.prometheus</li>
    <li>matrix-synapse</li>
    <li>matrix-synapse-plugins.matrix-synapse-mjolnir-antispam</li>
    <li>matrix-synapse-plugins.matrix-synapse-shared-secret-auth</li>
    <li>mautrix-facebook</li>
    <li>mautrix-googlechat</li>
    <li>mautrix-signal</li>
    <li>mautrix-telegram</li>
    <li>mlflow-server</li>
    <li>netbox</li>
    <li>peering-manager</li>
    <li>prometheus-bitcoin-exporter</li>
    <li>prometheus-kea-exporter</li>
    <li>prometheus-knot-exporter</li>
    <li>prometheus-pve-exporter</li>
    <li>prometheus-tor-exporter</li>
    <li>prometheus-xmpp-alerts</li>
    <li>python310Packages.aiohttp-openmetrics</li>
    <li>python310Packages.cairo-lang</li>
    <li>python310Packages.django-prometheus</li>
    <li>python310Packages.finitude</li>
    <li>python310Packages.flower</li>
    <li>python310Packages.gradient</li>
    <li>python310Packages.gradient-utils</li>
    <li>python310Packages.karton-dashboard</li>
    <li>python310Packages.mlflow</li>
    <li>python310Packages.prometheus-client</li>
    <li>python310Packages.prometheus-flask-exporter</li>
    <li>python310Packages.py-air-control-exporter</li>
    <li>python310Packages.ray</li>
    <li>python310Packages.uptime-kuma-monitor</li>
    <li>python311Packages.Nikola</li>
    <li>python311Packages.aiohttp-openmetrics</li>
    <li>python311Packages.augmax</li>
    <li>python311Packages.batchspawner</li>
    <li>python311Packages.django-prometheus</li>
    <li>python311Packages.dockerspawner</li>
    <li>python311Packages.einops</li>
    <li>python311Packages.finitude</li>
    <li>python311Packages.gradient-utils</li>
    <li>python311Packages.jupyter</li>
    <li>python311Packages.jupyter-lsp</li>
    <li>python311Packages.jupyter-server</li>
    <li>python311Packages.jupyterhub</li>
    <li>python311Packages.jupyterhub-ldapauthenticator</li>
    <li>python311Packages.jupyterhub-systemdspawner</li>
    <li>python311Packages.jupyterhub-tmpauthenticator</li>
    <li>python311Packages.jupyterlab</li>
    <li>python311Packages.jupyterlab-lsp</li>
    <li>python311Packages.jupyterlab_launcher</li>
    <li>python311Packages.jupyterlab_server</li>
    <li>python311Packages.jupytext</li>
    <li>python311Packages.karton-dashboard</li>
    <li>python311Packages.mkdocs-jupyter</li>
    <li>python311Packages.mlflow</li>
    <li>python311Packages.nbclassic</li>
    <li>python311Packages.nix-kernel</li>
    <li>python311Packages.notebook</li>
    <li>python311Packages.notebook-shim</li>
    <li>python311Packages.notedown</li>
    <li>python311Packages.oauthenticator</li>
    <li>python311Packages.prometheus-client</li>
    <li>python311Packages.prometheus-flask-exporter</li>
    <li>python311Packages.py-air-control-exporter</li>
    <li>python311Packages.uptime-kuma-monitor</li>
    <li>python311Packages.vega</li>
    <li>sourcehut.buildsrht</li>
    <li>sourcehut.coresrht</li>
    <li>sourcehut.gitsrht</li>
    <li>sourcehut.hgsrht</li>
    <li>sourcehut.hubsrht</li>
    <li>sourcehut.listssrht</li>
    <li>sourcehut.mansrht</li>
    <li>sourcehut.metasrht</li>
    <li>sourcehut.pastesrht</li>
    <li>sourcehut.todosrht</li>
    <li>tandoor-recipes</li>
    <li>xandikos</li>
  </ul>
</details>

mweinelt reviewed (changes requested):
The commit message should start with `python310Packages.prometheus-client`, since `prometheus-client` isn't a top-level attribute.

vamega commented on pkgs/development/python-modules/prometheus-client/default.nix:
```diff
@@ -16,7 +16,7 @@ buildPythonPackage rec {
     repo = "client_python";
     rev = "refs/tags/v${version}";
-    hash = "sha256-LabvQMNiRH4jclMnN0P4M3w25NQasNQEu1HWvRDQQ2o=";
+    sha256 = "sha256-FYQE0toy5VFKNVadSsxG/5NCRANYJOcVR4bGPrCAxvc=";
```
Thanks, will update that.  

I do see a oblique reference to the preference for `hash` over `sha256` at https://nixos.wiki/wiki/Nix_Hash
I had used `sha256` since when I tried to compute the value using `lib.fakeSha256` it wasn't able to determine the has value.

Is there a better way of identifying the correct hash when using the `hash` attribute?

vamega commented:
Thanks, updated the commit message, and switched back to using `hash`.

Out of curiosity, would `pythonPackages.prometheus-client` have been an acceptable prefix to the commit message, since this package is in both `python310Packages` and `python311Packages`?

mweinelt commented:
No, since `pythonPackages` refers to the python2 package set.

vamega commented:
I see but `python3Packages` would have been acceptable?

mweinelt commented:
Yes, it would. But there exists an ofBorg issue related to that: https://github.com/NixOS/ofborg/issues/577
